meos/src/general/temporal_modif.c

```diff
@@ -2117,7 +2117,7 @@ temporal_append_tsequence(const Temporal *temp, const TSequence *seq, bool expan
       return result;
     }
     case TSEQUENCE:
-      return (Temporal *) tsequence_append_tsequence((TSequence *) temp, seq,
+      return (Temporal *) tsequence_append_tsequence((const TSequence *) temp, seq,
```

Comment:
Here we should actually free `temp` before returning, since we are returning a new pointer.
This function is made to be used like this:
```c
temp = temporal_append_tsequence(temp, ...)
```
So we need to free the input if we return a newly allocated value

Reply:
The same goes for the `TINSTANT` case above actually

Reply:
Tests failed when I introduced this proposed change. As it's not part of the scope of this PR, will not include them here but rather in other PR if needed